pkg/utils/utils.go

```diff
@@ -600,6 +612,31 @@ func vfIsReady(pciAddr string) (netlink.Link, error) {
 	return vfLink, nil
 }
 
+func isValidAddress(address [6]byte) bool {
+	switch address {
+	case
+		[6]byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00},
+		[6]byte{0xFF, 0xFF, 0xFF, 0xFF, 0xFF, 0xFF}:
+		return false
+	}
+	return true
+}
+
+func getEffectiveMacAddress(hwAddr HardwareAddr) (HardwareAddr, error) {
+	newAddr = [6]byte{0x00, 0x00, 0x00, 0x00, 0x00, 0x00}
```

Comment:
why not use: `make([]byte, 6)` ? it will result in a slice of zero bytes ?

Reply:
The reason was that by making it more explicit, it would be easier to read.

Reply:
@adrianchiris Is that alright? 

> > Hi, I will fix up the commit message, once the CI is passing. Essentially VF effective mac address is all zero for some vendor NICs, so we have to set the VF effective MAC address so that it passes our QE tests. I will change the commit message to make sure that it reflects why this change is necessary.
> 
> Which vendor ? why is the MAC address zero for VF netdev, this seems like a kernel (driver) bug.

This is for Pensando DSC cards that we need to enable to get SRIOV network support on them. While other vendors do provide this within the drivers, this vendor does not. Therefore we must set it in the SRIOV network operator.